### PR TITLE
chore(nodejs): remove peer dep on @metamask/eslint-config

### PR DESCRIPTION
**Pull request by boneskull**
This config is _recommended_ to be used alongside `@metamask/eslint-config-nodejs`, but it is not actually referenced anywhere in the configuration.  It should not be a peer dependency.

### CODE REVIEW

**Comment by boneskull** [comment]
@legobeat No, because we cannot assume the consumer uses Yarn (v3?).  

The docs there explain what the field does, but not why it exists.  From what I can guess, if something _is_ in `peerDependencies` _and_ it's got an `optional` flag in `peerDependenciesMeta`, then Yarn won't complain if the peer dependency _isn't_ installed, so it's essentially a no-op; this assumes Yarn will refuse to automatically install peer dependencies (because if it _did_ automatically install them, I'm not seeing a use case for the field unless it's paired with some other Yarn-specific behavior).  npm _will_ install anything in `peerDependencies` _if it can_, and that's undesirable behavior if a package doesn't actually need those dependencies installed to work.

What'd make _more_ sense is a "suggested packages" field, which AFAIK no JS package manager supports, and that is fine by me. :D

**Comment by boneskull** [comment]
If I had any suggestion for the main consumers of this package to simplify consumption: create a "meta" eslint config which depends directly on all these packages and just re-exports all of them as a blob.  alternatively, make a null plugin with a few prerolled configs in it (`recommended`, `node`, etc.).

**Comment by boneskull** [comment]
@legobeat lol, must be new.  

npm's docs are wrong.  It doesn't warn about a missing peer dependency; it just installs it.  I'm not sure what it'd do with this field, but regardless, listing `@metamask/eslint-config` in `peerDependencies` would be incorrect because it is not an optional dependency; it's not a dependency _at all_.

**Comment by boneskull** [comment]
> This allows you to integrate and interact with a variety of host packages without requiring all of them to be installed

okay, assuming that's valid (I have my doubts; I wonder what querying the registry would show) the "host" package for this workspace is `eslint`, so that seems semantically incorrect

**Comment by mcmire** [comment]
> listing `@metamask/eslint-config` in `peerDependencies` would be incorrect because it is not an optional dependency; it's not a dependency at all.

It's not a dependency in the strict sense. You are right about that. You don't need `eslint-config` to install any of the `eslint-config-*` packages.

That said, I don't know if it's correct to think of `eslint-config` as a "suggested package". There is a relevant and important connection between the specific and the base packages. If you _do_ install `eslint-config` alongside one of the `eslint-config-*` — and we do within MetaMask, as we use all of these together — then we need some way of indicating which version of `eslint-config` we expect you to have. We want to safeguard against projects upgrading an `eslint-config-*` without also upgrading `eslint-config` itself. So, we use `peerDependency` to indicate this.

As stated above, however, `eslint-config` is not a requirement, so it doesn't make sense for it to be a _required_ peer dependency. Instead, it should be an optional peer dependency. This way, if `eslint-config` is not installed, it doesn't complain, but if it _is_ installed, then it will complain if the version isn't right.

As for the question of support for optional peer dependencies among package managers, it seems that the latest versions of NPM, Yarn, and PNPM do; but Yarn v1 does not.

To test this, I created a project with a single dependency, `@twind/react`, which has an [optional peer dependency on `typescript`](https://github.com/tw-in-js/twind-react/blob/main/package.json#L42):

- With NPM 9.8.1, I ran `npm install`, and I did not get a warning about `typescript` not being installed. I also checked `node_modules` and `typescript` was not present. Then I added `typescript` ^5.0.0, which does not match the peer dependency, and I got the following error:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @twind/react@0.0.4
npm ERR! Found: typescript@5.1.6
npm ERR! node_modules/typescript
npm ERR!   typescript@"^5.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peerOptional typescript@"^4.1.0" from @twind/react@0.0.4
npm ERR! node_modules/@twind/react
npm ERR!   @twind/react@"^0.0.4" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: typescript@4.9.5
npm ERR! node_modules/typescript
npm ERR!   peerOptional typescript@"^4.1.0" from @twind/react@0.0.4
npm ERR!   node_modules/@twind/react
npm ERR!     @twind/react@"^0.0.4" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

- With Yarn 1.22.19, I ran `yarn install`, and I did not get a warning about `typescript` not being installed. I also checked `node_modules` and `typescript` was not present. Then I added `typescript` ^5.0.0, which does not match the peer dependency, and I did not get an error.

- With Yarn 3.6.1, I ran `yarn install`, and I did not get a warning about `typescript` not being installed. I also ran `yarn why typescript` and did not get any output. Then I added `typescript` ^5.0.0, which does not match the peer dependency, and I got the following warning:

```
➤ YN0060: │ my-project@workspace:. provides typescript (p26643) with version 5.1.6, which doesn't satisfy what @twind/react requests
```

- With `pnpm` 8.6.10, I ran `pnpm install`, and I did not get a warning about `typescript` not being installed. I also checked `node_modules` and `typescript` was not present. Then I added `typescript` ^5.0.0, which does not match the peer dependency, and I got the following warning:

```
 WARN  Issues with peer dependencies found
.
├─┬ @twind/react 0.0.4
│ └── ✕ unmet peer typescript@^4.1.0: found 5.0.2
└─┬ twind 0.16.19
  └── ✕ unmet peer typescript@^4.1.0: found 5.0.2
```

In summary, I believe that peer dependency is the right tool here, as it aligns peers, which in this case are the `eslint-config` package and `eslint-config-*` packages. However, making `eslint-config` required is wrong, and we should make it optional instead. This essentially makes the check on `eslint-config` "nice to have", because Yarn v1 does not honor optional peer dependencies. This is an acceptable compromise to me considering that it is a legacy version.

What do you think about this @boneskull ?

**Comment by boneskull** [comment]
npm 9 will still install it automatically whether it's optional or not, which is undesirable. however, if there's a version conflict, npm will throw an error. 

that said, because lavamoat is consuming this package _in a way which is unintended_--as you've explained--I think lavamoat should probably just break the dep and copy the rules.
